src/main/java/gregtech/api/metatileentity/SimpleGeneratorMetaTileEntity.java

```diff
@@ -15,24 +15,29 @@
 import gregtech.api.gui.widgets.ImageWidget;
 import gregtech.api.gui.widgets.SlotWidget;
 import gregtech.api.gui.widgets.TankWidget;
+import gregtech.api.metatileentity.sound.ISoundCreator;
+import gregtech.api.metatileentity.sound.PositionedSoundMTE;
 import gregtech.api.recipes.machines.FuelRecipeMap;
 import gregtech.api.render.OrientedOverlayRenderer;
 import gregtech.api.render.Textures;
 import gregtech.api.util.PipelineUtil;
+import gregtech.common.ConfigHolder;
+import net.minecraft.client.Minecraft;
 import net.minecraft.client.resources.I18n;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.NonNullList;
 import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundCategory;
 import net.minecraft.world.World;
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.items.ItemStackHandler;
 
 import javax.annotation.Nullable;
 import java.util.List;
 
-public class SimpleGeneratorMetaTileEntity extends TieredMetaTileEntity {
+public class SimpleGeneratorMetaTileEntity extends TieredMetaTileEntity implements ISoundCreator {
```

Comment:
Why do you implement `ISoundCreator` on the various MTE subclasses instead of directly on MetaTileEntity?

Reply:
As not all MTEs create sound, and many of them source their recipe maps from different places, I found it easier to just implement them on each of the subclasses that required the implementation rather than to generalize the implementation in MetaTileEntity and override it several times.